pkg/daemon/ceph/osd/daemon.go

```diff
@@ -415,6 +415,25 @@ func getAvailableDevices(context *clusterd.Context, agent *OsdAgent) (*DeviceOsd
 				}
 				matchedDevice = desiredDevice
 
+				if agent.pvcBacked {
+					crushDeviceClass := os.Getenv(oposd.CrushDeviceClassVarName)
+					if crushDeviceClass != "" {
+						matchedDevice.DeviceClass = crushDeviceClass
+					}
+				}
```

Comment:
Won't this be overridden by the code below? It should be the same condition block I believe.

Reply:
No, below condition on != empty string, if we set it there - below code will not run

Reply:
Oops my bad!